sqlmesh/core/loader.py

```diff
@@ -165,18 +188,25 @@ def _load_sql_models(
         self, macros: MacroRegistry, hooks: HookRegistry
     ) -> UniqueKeyDict[str, Model]:
         """Loads the sql models into a Dict"""
+        model_cache = ModelCache(self._context.path / c.CACHE)
+
         models: UniqueKeyDict = UniqueKeyDict("models")
         for context_path, config in self._context.configs.items():
-            for path in self._glob_paths(context_path / c.MODELS, config=config, extension=".sql"):
+            models_path = context_path / c.MODELS
+            for path in self._glob_paths(models_path, config=config, extension=".sql"):
                 self._track_file(path)
-                with open(path, "r", encoding="utf-8") as file:
-                    try:
-                        expressions = parse(
-                            file.read(), default_dialect=config.model_defaults.dialect
-                        )
-                    except SqlglotError as ex:
-                        raise ConfigError(f"Failed to parse a model definition at '{path}': {ex}")
-                    model = load_model(
+
+                def _load() -> Model:
```

Comment:
What is the cost of defining this method on every inner loop? From what I can tell, the method can be moved out of the inner loop if path is passed in as a param.

Reply:
I don't think the cost matters in this case.